realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1492,6 +1493,11 @@ public Long call() throws Exception {
 
                         return handoverTableViewPointer;
 
+                    } catch (BadVersionException e) {
+                        // In some rare race conditions, this can happen. In that case, just ignore the error.
+                        RealmLog.d("findAllAsync handover could not complete due to a BadVersionException. " +
```

Comment:
Ah, sorry, missed that.
Still don't understand what you mean with "Should" though?